data/scripts.js

```diff
@@ -3644,9 +3644,20 @@ exports.BattleScripts = {
 		var pool = Object.keys(sets).randomize();
 		var ranks = {'~':'admins', '&':'leaders', '@':'mods', '%':'drivers', '+':'voices'};
 		var levels = {'~':99, '&':97, '@':96, '%':96, '+':95};
-		for (var i = 0; i < 6; i++) {
-			var rank = pool[i].charAt(0);
+		var count = 0;
+		var i = -1;
```

Comment:
`i` is the set index, `count` is the number of mons added to the team. If some mons are rejected due to species clause `i` will end up being larger than `count`.